crates/modelardb_compression/src/models/swing.rs

```diff
@@ -193,12 +213,13 @@ impl Swing {
     /// only require `size_of::<Value>` while the slope and intercept generally
     /// must be [`f64`] to be precise enough.
     pub fn model(self) -> (Value, Value) {
-        // TODO: use the function with the minimum error as specified in the Swing and Slide paper.
-        let average_slope = (self.lower_bound_slope + self.upper_bound_slope) / 2.0;
-        let average_intercept = (self.lower_bound_intercept + self.upper_bound_intercept) / 2.0;
-        let first_value = average_slope * self.start_time as f64 + average_intercept;
-        let last_value = average_slope * self.end_time as f64 + average_intercept;
-        (first_value as Value, last_value as Value)
+        // Implementation of the function with the minimized MSE as specified in the Swing and Slide paper.
+        // Equation 5 and 6
+        let projected_slope = self.slope_with_minimum_mse.0 / self.slope_with_minimum_mse.1;
+        let optimal_slope =  self.lower_bound_slope.max(projected_slope.min(self.upper_bound_slope));
```

Comment:
It seems there is an extra space after `=`. Maybe use `Rustfmt` to be sure all issues like this is fixed.

Reply:
Fixed.